src/zkp/pifac.rs

```diff
@@ -329,136 +332,143 @@ fn sqrt(num: &BigNumber) -> BigNumber {
+            let setup_param = VerifiedRingPedersen::gen(&mut rng2, &())?;
+            let incorrect_startup_params = CommonInput::new(&setup_param, input.modulus);
             assert!(proof
-                .verify(&incorrect_startup_params, &(), &mut transcript)
+                .verify(incorrect_startup_params, &(), &mut transcript())
                 .is_err());
-        }
-        {
-            let mut transcript = Transcript::new(b"PiFac Test");
-            let (not_p0, not_q0) = prime_gen::get_prime_pair_from_pool_insecure(&mut rng).unwrap();
+            Ok(())
+        };
+        with_random_no_small_factors_proof(&mut rng, test_code2)?;
+
+        let test_code3 = |input: CommonInput, _proof: PiFacProof| {
+            let (not_p0, not_q0) = prime_gen::get_prime_pair_from_pool_insecure(&mut rng2).unwrap();
             let incorrect_factors = PiFacProof::prove(
-                &input,
-                &ProverSecret::new(&not_p0, &not_q0),
+                input,
+                ProverSecret::new(&not_p0, &not_q0),
                 &(),
-                &mut transcript,
-                &mut rng,
+                &mut transcript(),
+                &mut rng2,
             )?;
-            let mut transcript = Transcript::new(b"PiFac Test");
+
             assert!(incorrect_factors
-                .verify(&input, &(), &mut transcript)
+                .verify(input, &(), &mut transcript())
                 .is_err());
 
-            let mut transcript = Transcript::new(b"PiFac Test");
             let small_p = BigNumber::from(7u64);
             let small_q = BigNumber::from(11u64);
-            let setup_params = VerifiedRingPedersen::gen(&mut rng, &())?;
-            let small_input = CommonInput::new(&setup_params, &(&small_p * &small_q));
+            let setup_params = VerifiedRingPedersen::gen(&mut rng2, &())?;
+            let modulus = &small_p * &small_q;
+            let small_input = CommonInput::new(&setup_params, &modulus);
             let small_proof = PiFacProof::prove(
-                &input,
-                &ProverSecret::new(&small_p, &small_q),
+                input,
+                ProverSecret::new(&small_p, &small_q),
                 &(),
-                &mut transcript,
-                &mut rng,
+                &mut transcript(),
+                &mut rng2,
             )?;
-            let mut transcript = Transcript::new(b"PiFac Test");
+
             assert!(small_proof
-                .verify(&small_input, &(), &mut transcript)
+                .verify(small_input, &(), &mut transcript())
                 .is_err());
 
-            let mut transcript = Transcript::new(b"PiFac Test");
-            let regular_sized_q = prime_gen::try_get_prime_from_pool_insecure(&mut rng).unwrap();
-            let mixed_input = CommonInput::new(&setup_params, &(&small_p * &regular_sized_q));
+            let regular_sized_q = prime_gen::try_get_prime_from_pool_insecure(&mut rng2).unwrap();
```

Comment:
Same other note as before: this should also test the case where exactly one of the factors is out of range. I guess we do it for small p but not small q...?